### PR TITLE
Replace demo READMEs with links to Read the Docs

### PR DESCRIPTION
**Pull request by huonw**
This augments the `demo_table.py` script (and thus renames it to `demo_indexing.py`) to support adding:

- a link to the corresponding Read the Docs page for the task (e.g. `demos/node-classification/README.md` points to https://stellargraph.readthedocs.io/en/latest/demos/node-classification/index.html )
- a listing of each demo within the subfolder, including both a relative link ("open here") and a link to the Read the Docs version ("rendered")

This is designed to provide good guidance for a user to get to our nicely displayed demos on Read the Docs, and also provide them with basic indexing when opening the READMEs while actually running a demo (either in a text editor etc., or in Jupyter Lab).

The other contents is removed except for the title, since that content has been moved to Read the Docs (#1391).

Finally, the `demos/link-prediction/random-walks/utils/node2vec/README.md` README was removed. It is inherited directly from https://github.com/aditya-grover/node2vec and doesn't convey relevant content, and there's a separate LICENSE file that fulfils our legal obligation with that included code.

Example: https://github.com/stellargraph/stellargraph/tree/bugfix/1419-demo-readmes/demos/node-classification

View in Jupyter Lab (e.g. on Binder):  
image

Top level entry point: https://github.com/stellargraph/stellargraph/tree/bugfix/1419-demo-readmes/demos

See: #1419

### CODE REVIEW

**Comment by timpitman** [comment]
The output README's look good, my only suggestion is potentially the links could be something different than "rendered" and "open here"?  something like "demo in documentation", "GitHub source"?

**Comment by huonw** [comment]
I like the idea of the whole title being RtD.

I chose "open here" because it's mainly designed for someone viewing the markdown file locally in a Jupyter Lab; by clicking that link they jump straight to the notebook in that jupyter lab instance.

**Comment by huonw** [comment]
Maybe just "source" works for that too, though? I'll go with that.

**Comment by timpitman** [comment]
Looks good now!

File: scripts/demo_indexing.py [hunk 1]
```diff
@@ -17,16 +17,27 @@
...
 import os.path
+import re
 import subprocess
 import sys
 import textwrap
 
 HTML_INDENT = 2
 LINK_DEFAULT_TEXT = "demo"
 TRUE_TEXT = "yes"
+# FIXME: this should point to stable, not latest
```
[the remark this answers is not in the document]
Comment (huonw) [reply scripts/demo_indexing.py]: Good catch; I was intending to do it before landing, so I've done it now. We can land this tomorrow to minimise the time when links are broken?

**Pull request merged**